### ISSUE TITLE
add getAllLayersAndTables() to arcgis-rest-feture-layer

### ISSUE THREAD

**Issue by tomwayson** [Esri/arcgis-rest-js#874]
```ts
/** begin move this to rest-js */
/**
 * Match the "service" part of the url
 */
const serviceRegex = new RegExp(/.+(?:map|feature|image)server/i);
/**
 * Return the service url. If not matched, returns what was passed in
 */
export function parseServiceUrl (url: string) {
  const match = url.match(serviceRegex);
  if (match) {
    return match[0];
  } else {
    return stripQueryString(url);
  }
}
/**
 * Match the arcGIS server "root"
 * which always ends with /rest
 */
const serverRegex = new RegExp(/(\/rest).+/i);
/**
 * Return the server url, if not matched, returns null
 */
export function parseServerUrl (url: string) {
  const match = url.match(serverRegex);
  if (match) {
    return url.replace(serverRegex, '$1');
  } else {
    return stripQueryString(url);
  }
}

function stripQueryString (url: string) {
  const stripped = url.split('?')[0];
  return cleanUrl(stripped);
}

// should we expand this to support other valid params?
// see: https://developers.arcgis.com/rest/services-reference/all-layers-and-tables.htm
const getAllLayersAndTables = (options: IGetLayerOptions) => {
  const { url, ...requestOptions } = options;
  const layersUrl = `${parseServiceUrl(url)}/layers`;
  return request(layersUrl, requestOptions);
}
/** end move to rest-js */
```

**Comment by thomas-hervey** [comment]
Tracking this in dc/hub: https://devtopia.esri.com/dc/hub/issues/1406
@sonofflynn89

**Comment by tomwayson** [comment]
resolved in #876
